apps/ledger-live-mobile/src/components/DeviceAction/index.tsx

```diff
@@ -431,35 +427,34 @@ export function DeviceActionDefaultRendering<R, H extends Status, P>({
   }
 
   if (completeExchangeStarted && !completeExchangeResult && !completeExchangeError) {
-    return renderExchange({
-      exchangeType: (request as { exchangeType: number })?.exchangeType,
-      t,
-      device: device!,
+    const swapRequest = {
+      ...request,
+      colors,
+      selectedDevice,
       theme,
-    });
-  }
-
-  if (initSwapRequested && !initSwapResult && !initSwapError) {
-    const req = request as {
-      device: Device;
+      walletState,
+      settingsState,
+      estimatedFees: status.estimatedFees,
+    } as {
+      selectedDevice: Device;
+      provider: string;
       transaction: Transaction;
       exchangeRate: ExchangeRate;
       exchange: ExchangeSwap;
+      colors: T["colors"];
```

Comment:
Shouldn't it be this instead here ?
```suggestion
      colors: Theme["colors"];
```

Reply:
fixed